core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/arrow/ArrowNltkSentiment.java

```diff
@@ -0,0 +1,274 @@
+package edu.uci.ics.texera.dataflow.arrow;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import edu.uci.ics.texera.api.constants.ErrorMessages;
+import edu.uci.ics.texera.api.constants.SchemaConstants;
+import edu.uci.ics.texera.api.constants.DataConstants.TexeraProject;
+import edu.uci.ics.texera.api.dataflow.IOperator;
+import edu.uci.ics.texera.api.exception.DataflowException;
+import edu.uci.ics.texera.api.exception.TexeraException;
+import edu.uci.ics.texera.api.field.IField;
+import edu.uci.ics.texera.api.field.IntegerField;
+import edu.uci.ics.texera.api.schema.AttributeType;
+import edu.uci.ics.texera.api.schema.Schema;
+import edu.uci.ics.texera.api.tuple.Tuple;
+import edu.uci.ics.texera.api.utils.Utils;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.*;
+import org.apache.arrow.vector.ipc.ArrowFileReader;
+import org.apache.arrow.vector.ipc.SeekableReadChannel;
+import org.apache.arrow.vector.ipc.message.ArrowBlock;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+
+import static java.util.Arrays.asList;
+
+public class ArrowNltkSentiment implements IOperator {
+    private final ArrowNltkSentimentPredicate predicate;
+    private IOperator inputOperator;
+    private Schema outputSchema;
+    private List<Tuple> batchedTupleBuffer;
+    HashMap<String, Integer> idClassMap;
+    private int cursor = CLOSED;
+    private String picklePath = null;
+    private final int batchSize;
+    private final int chunkSize;
+    private final String inputAttributeName;
+    private final String outputAttributeName;
+
+    private final static String PYTHON = "python";
+    private final static String PYTHON_SCRIPT = Utils.getResourcePath(
+            "arrow_for_nltk_sentiment.py", TexeraProject.TEXERA_DATAFLOW).toString();
+    private final static String TO_PYTHON_FILE = Utils.getResourcePath(
+            "temp-to-python.arrow", TexeraProject.TEXERA_DATAFLOW).toString();
+    private final static String FROM_PYTHON_FILE = Utils.getResourcePath(
+            "temp-from-python.arrow", TexeraProject.TEXERA_DATAFLOW).toString();
+
+    private final static char SEPARATOR = ',';
+    private final static char QUOTE_CHAR = '"';
+
+    private final static org.apache.arrow.vector.types.pojo.Schema tupleToPythonSchema =
```

Comment:
I think for this PR and this operator, we can assume a fixed schema of passing (ID, text) to python, and get (ID, int) back from python. For next step, we need to have a more general implementation that supports arbitrary tuple and schema.

Reply:
Yes, that is the plan.